Files/powerGate/Modules/Communication.psm1

```diff
@@ -2,12 +2,31 @@
 Import-Module powerGate
 
 #TODO: configure the powerGate server url and port
-$powerGateServerName = $ENV:Computername
+$powerGateServerName = getRelatedPGServerName
 $powerGateServerPort = "8080"
+$vaultDefinitions = @{
+	"TEST" = "TestVault";
+	"PROD" = "ProductiveVault"
+}
+$PGServerDefinitions = @{
+	"TEST" = "TestERP";
+	"PROD" = "ProductiveERP"
+}
 $powerGateServerErpPluginUrl = "http://$($powerGateServerName):$($powerGateServerPort)/coolOrange/ErpServices"
 #$powerGateServerErpPluginUrl = "http://$($powerGateServerName):$($powerGateServerPort)/coolOrange/DynamicsNav"
 # Dynamics NAV 2017 Plugin available here: https://github.com/coolOrangeLabs/powergate-dynamics-nav-sample/releases
 
+function getRelatedPGServerName {
+	switch ($vaultConnection.Vault) {
+		$VaultDefinitions["TEST"] {
+			return $PGServerDefinitions["TEST"]
+		}
+		$VaultDefinitions["PROD"] {
+			return $PGServerDefinitions["PROD"]
+		}
+		Default { return $null}
```

Comment:
It means when the currently connected vault is not "TEST" or "PROD" then it returns no powerGateServer.
This is critical if the connected VAULT is not defined!

 **ToDO** there should pop up an error to the user like _"Could not connect to any ERP because the current connected VAULT 'XY' is not mapped in the configuration file for any ERP. Change the configuration and restart vault!"_